### PR TITLE
switch to paper for compiling

### PR DESCRIPTION
**Pull request by jerzean**
add velocity check/support
Update Readme

>Pull requests should target **master**.
If your pull requests are not ready to be reviewed, please either label it as `WIP` or open a draft pull request.

## General Checks
Place an X in the boxes that you have completed for this pull request.

- [X] I have linked the ticket associated with this pull request.
- [X] I have fully tested my pull request.

## PR Notes
Add some general notes here about what your PR solves for.

### CODE REVIEW

File: src/main/java/com/plugish/woominecraft/WooMinecraft.java [hunk 1]
```diff
@@ -69,13 +71,43 @@ public void onEnable() {
...
 		// Log when plugin is fully enabled ( setup complete ).
 		getLogger().info( this.getLang( "log.enabled" ) );
-		//check bungeecord mode/offline mode
-		if (!Bukkit.getOnlineMode() && !Bukkit.spigot().getConfig().getBoolean("settings.bungeecord")) {
-			getLogger().severe(String.valueOf(Bukkit.spigot().getConfig().getBoolean("settings.bungeecord")));
-			getLogger().severe("WooMinecraft doesn't support offLine mode");
-			Bukkit.getPluginManager().disablePlugin(this);
+		//check bungeecord/velocity mode/offline mode
+		try {
+			getLogger().info(Bukkit.getServer().getName());
```
Comment (JayWood) [inline src/main/java/com/plugish/woominecraft/WooMinecraft.java:76]: Repeating usages of `Bukkit.getServer().getName()` let's make this a variable. Keep the code `DRY`

File: src/main/java/com/plugish/woominecraft/WooMinecraft.java [hunk 2]
```diff
@@ -69,13 +71,43 @@ public void onEnable() {
...
+			//check if the server is spigot/bukkit
+			if (Bukkit.getServer().getName().equals("CraftBukkit"))
```
Comment (JayWood) [inline src/main/java/com/plugish/woominecraft/WooMinecraft.java:78]: Missing a bracket here?

File: src/main/java/com/plugish/woominecraft/WooMinecraft.java [hunk 3]
```diff
@@ -69,13 +71,43 @@ public void onEnable() {
...
+				if (!Bukkit.getOnlineMode() && !Bukkit.spigot().getConfig().getBoolean("settings.bungeecord")) {
+					getLogger().severe("BungeeCord: "+Bukkit.spigot().getConfig().getBoolean("settings.bungeecord"));
```
Comment (JayWood) [inline src/main/java/com/plugish/woominecraft/WooMinecraft.java:80]: Given how many `getLogger.xxx` calls we have here, and the depth of the nesting, I'd either like to see this nesting simplified ( by creating separate methods ) or maybe create a logging class entirely and calling it here. Perhaps this is the web developer in me but I'm not used to seeing such deeply nested IF statements.

File: src/main/java/com/plugish/woominecraft/WooMinecraft.java [hunk 4]
```diff
@@ -69,13 +71,43 @@ public void onEnable() {
...
+					getLogger().severe("Online-Mode:"+Bukkit.getOnlineMode());
+					getLogger().severe("WooMinecraft doesn't support offLine mode");
+					getLogger().severe("If you have a bungeecord server, please set Bungeecord to true and Online-mode to false");
+					Bukkit.getPluginManager().disablePlugin(this);
+					//check if the server is Paper
+				} else if (Bukkit.getServer().getName().equals("Paper")) {
+					// check if velocity is enabled and if its in online mode
+					if (!Bukkit.getOnlineMode() && !Bukkit.spigot().getPaperConfig().getBoolean("settings.velocity-support.enabled")) {
```
Comment (JayWood) [inline src/main/java/com/plugish/woominecraft/WooMinecraft.java:88]: Multiple uses of `Bukkit.getOnlineMode()` - needs to be a variable for re-use.

File: src/main/java/com/plugish/woominecraft/WooCommand.java [hunk 5]
```diff
@@ -157,13 +157,18 @@ private void pingSubcommand(CommandSender sender) {
...
+                    //Do not output store keys to console, can open up potential issues
                     WooMinecraft.instance.getLogger().info(plugin.getConfig().getString("url"));
                 }
             }
         });
     }
     private void debugSubcommand(CommandSender sender) {
+        if (!sender.hasPermission("woo.admin")) {
+            String msg = chatPrefix + ChatColor.translateAlternateColorCodes('&', plugin.getLang("general.not_authorized"));
+            sender.sendMessage(msg);
```
Comment (JayWood) [inline src/main/java/com/plugish/woominecraft/WooCommand.java:169]: Just simplify this, no need for a vairable:
`sender.sendMessage( chatPrefix + ChatColor.translateAlternateColorCodes( '&', plugin.getLang( "general.not_authorized" ) ) );`

**Review by JayWood — CHANGES_REQUESTED** [review]

**Comment by JayWood** [comment]
Build also failed, need to make sure it passes. Remember you can click on details, then the `the build` link on the resulting page to see full build details.

**Comment by OverBrave** [comment]
When can this compilation be published?

**Comment by JayWood** [comment]
When the build is not failing. From what I can tell of the build log I'm getting this:
`[ERROR] /home/travis/build/WooMinecraft/WooMinecraft/src/main/java/com/plugish/woominecraft/WooMinecraft.java:[18,17] error: cannot access Bukkit`

https://app.travis-ci.com/github/WooMinecraft/WooMinecraft/builds/244297003#L710 Not sure why as I wasn't able to see any changes in his code that should have caused this. I'll retarget the branch this week and merge to a feature branch and see if I can fix the issue.

**Pull request merged**